### PR TITLE
Fix compilation warnings when IPv4 and IPv6 are disabled.

### PR DESCRIPTION
**Pull request by rveerama1**
Packet socket sample doesn't need CONFIG_NET_IPV4, CONFIG_NET_IPV6, but we can not build without it. This PR fixes compilation warnings if you disable IPv4 and IPv6.

### CODE REVIEW

**Review by tbursztyka — APPROVED** [review]
seems fine to me (as long as CI passes).
Let's see one day if we can reduce the ifdef (not this PR's role).

File: subsys/net/ip/net_context.c [hunk 1]
```diff
@@ -52,6 +52,7 @@ static struct net_context contexts[NET_MAX_CONTEXT];
  */
```
Comment (jukkar) [inline subsys/net/ip/net_context.c]: The "net: Fix compilation warnings" is in a way correct subject, but I think it does not reflect the spirit of this patch. Better subject would be "net: Allow disabling IPv4, IPv6, UDP and TCP simultaneously". The body then tells that we fix the complation warnings.

**Review by jukkar — CHANGES_REQUESTED** [review]
Looks good, just a minor nit-pick about the commit subject

File: subsys/net/ip/net_context.c (hunk 1 above)
Comment (rveerama1) [reply subsys/net/ip/net_context.c]: I thought it wouldn't fit long text like that. Tried to shrink it and the result was not clear. Thanks, updated it.

**Review by jukkar — APPROVED** [review]
LGTM

**Pull request merged**